### PR TITLE
Add blur transformation

### LINKED ISSUE #389: Feature-request: Blur as transformation
What do you think if we add blur as new transormation?(http://php.net/manual/de/imagick.blurimage.php)

### PR DESCRIPTION
**Pull request by kbrabrand**
This PR implements blur as a transformation in Imbo as requested by @sgasser in #389. As per the discussion in the issue, channel support has not been implemented but apart from that all the different blur types and options supported by imagick has been included.

Closes #389.

### CODE REVIEW

File: docs/usage/image-transformations.rst [hunk 1]
```diff
@@ -20,6 +20,44 @@ This transformation will auto rotate the image based on EXIF data stored in the
...
+
+Blur the image - ``t[]=blur``
+-----------------------------------
+
+This transformation can be used to blur the image.
+
+**Parameters:**
+
+``mode``
+    The blur type (optional). Possible values are:
```
Comment (rexxars) [inline docs/usage/image-transformations.rst]: Optional, but which one is default?

File: library/Imbo/Image/Transformation/Blur.php [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+     * Add motion blur to the image
+     *
+     * @param EventInterface $event The event instance
+     */
+    private function motionBlur(EventInterface $event) {
+        $params = $event->getArgument('params');
+
+        $this->checkRequiredParams($params, ['radius', 'sigma', 'angle']);
+
+        if (isset($params['radius'])) {
```
Comment (rexxars) [inline library/Imbo/Image/Transformation/Blur.php]: Can't see that radius is actually used here?

File: library/Imbo/Image/Transformation/Blur.php [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            $radius = (float) $params['radius'];
+        }
+
+        if (isset($params['sigma'])) {
+            $sigma = (float) $params['sigma'];
+        }
+
+        try {
+            if ($adaptive) {
+                $this->imagick->adaptiveBlurImage($radius, $sigma);
```
Comment (rexxars) [inline library/Imbo/Image/Transformation/Blur.php]: If either radius or sigma is not set in the params, won't this cause a notice error due to the variable not being defined?

Comment (kbrabrand) [inline library/Imbo/Image/Transformation/Blur.php]: `checkRequiredParams` will throw an exception before the values are used.

File: library/Imbo/Image/Transformation/Blur.php (hunk 2 above)
Comment (kbrabrand) [inline library/Imbo/Image/Transformation/Blur.php]: I managed to add a typo while adding tests. :tired_face: Good job.. Fixed now

File: library/Imbo/Image/Transformation/Blur.php (hunk 3 above)
Comment (rexxars) [inline library/Imbo/Image/Transformation/Blur.php]: Then why do the isset check?

Comment (kbrabrand) [inline library/Imbo/Image/Transformation/Blur.php]: Good point.

**Pull request merged**